lib/max_subarray.py

```diff
@@ -2,11 +2,20 @@
 def max_sub_array(nums):
     """ Returns the max subarray of the given list of numbers.
         Returns 0 if  nums is None or an empty list.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)
+        Space Complexity: O(n)
```

Comment:
✨ Nice work, however space complexity is O(1) here because you are not actually making any new data structures.